src/app/(profile)/_actions/countProfileView.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  getProfileById,
+  updateProfileById,
+} from '@/backend/profile/profile.service'
+import { withSentry } from '@/utils/errHandling'
+import { Prisma } from '@prisma/client'
+
+export const countProfileViews = withSentry(
+  async (profileId: string, payload: Prisma.ProfileUpdateInput) => {
+    const foundProfile = await getProfileById(profileId)
+
+    if (!foundProfile) {
+      throw new Error('Profile not found')
+    }
+
+    payload.viewCount = foundProfile.viewCount + 1
+
```

Comment:
we should validate that when authorized user is owner of the profile it should not be incremented